app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt

```diff
@@ -50,9 +50,19 @@ class SellPayloadPresenter(
     private var rawUserSelectedAmount: String = emptyString()
     private val userSelectedAmount: BigDecimal
         get() = rawUserSelectedAmount.toBigDecimalOrZero()
-    private var currentFiat: SellTransactionFiatCurrency = SellTransactionFiatCurrency.USD
     private var tokenPrice: BigDecimal = BigDecimal.ZERO
 
+    private lateinit var tokenCurrencyMode: CurrencyMode.Token
+    private var fiatCurrencyMode: CurrencyMode.Fiat = CurrencyMode.Fiat.Usd
+    private lateinit var selectedCurrencyMode: CurrencyMode
+    private val currencyModeToSwitch: CurrencyMode
+        get() = when (selectedCurrencyMode) {
+            is CurrencyMode.Token -> fiatCurrencyMode
+            is CurrencyMode.Fiat -> tokenCurrencyMode
+        }
+
+    private lateinit var viewState: ViewState
```

Comment:
you have a lot of global variables. 
Looks like you need some separate class for calculation manipulations

Reply:
we can do better then lateinit and lots of variables with Empty state and refactoring
do you mind if i create a tech debt task for this?